python/xorbits/_mars/services/session/supervisor/core.py

```diff
@@ -45,9 +45,25 @@ async def __pre_destroy__(self):
             *[mo.destroy_actor(ref) for ref in self._session_refs.values()]
         )
 
-    async def create_session(self, session_id: str, create_services: bool = True):
+    async def _get_namespace(self) -> Optional[str]:
+        envs = await self._cluster_api.get_nodes_info(env=True)
+        for node in envs:
+            env = envs[node].get("env", {})
+            if "k8s_namespace" in env:
+                return env["k8s_namespace"]
```

Comment:
Specify `role=NodeRole.SUPERVISOR` to get supervisor's info which is more efficient.